src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.vb

```diff
@@ -433,6 +434,363 @@ End Class
+            comp.AssertTheseDiagnostics(<expected><![CDATA[
+BC37228: The field has multiple distinct constant values.
+    <DateTimeConstant(42)>
+     ~~~~~~~~~~~~~~~~~~~~
+]]></expected>)
+
+        End Sub
+
+        <WorkItem(217740, "https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems?id=217740")>
+        <Fact()>
+        Public Sub DateTimeConstantAttributeReferencedViaRef()
+            Dim source1 =
+<compilation>
+    <file name="attr.vb"><![CDATA[
+Imports System
+Imports System.Runtime.CompilerServices
+
+Public Class Bar
+    Public Sub Method(<DateTimeConstant(-1)> p1 As DateTime)
+    End Sub
+End Class
+]]>
+    </file>
+</compilation>
+
+            Dim source2 =
+<compilation>
+    <file name="attr.vb"><![CDATA[
+Public Class Consumer
+    Public Shared Sub Main()
+        Dim test = New Bar()
+        test.Method()
+    End Sub
+End Class
+]]>
+    </file>
+</compilation>
+
+            Dim libComp = CreateCompilationWithMscorlib(source1)
+            Dim libCompRef = New VisualBasicCompilationReference(libComp)
+
+            Dim comp2 = CreateCompilationWithMscorlib(source2, references:={libCompRef})
+            AssertTheseDiagnostics(comp2,
+                                   <expected><![CDATA[
+BC30455: Argument not specified for parameter 'p1' of 'Public Sub Method(p1 As Date)'.
+        test.Method()
+             ~~~~~~
+]]></expected>)
+
+            Dim libAssemblyRef = libComp.EmitToImageReference()
+            Dim comp3 = CreateCompilationWithMscorlib(source2, references:={libAssemblyRef})
+            AssertTheseDiagnostics(comp3,
+                <expected><![CDATA[
+BC30455: Argument not specified for parameter 'p1' of 'Public Sub Method(p1 As Date)'.
+        test.Method()
+             ~~~~~~
+]]></expected>)
+        End Sub
+
+        <WorkItem(217740, "https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_workitems?id=217740")>
+        <Fact()>
+        Public Sub LoadingDateTimeConstantWithBadValueOnField()
+            Dim ilSource = <![CDATA[
+.class public auto ansi C
+       extends [mscorlib]System.Object
+{
+  .field public static initonly valuetype [mscorlib]System.DateTime F
+  .custom instance void [mscorlib]System.Runtime.CompilerServices.DateTimeConstantAttribute::.ctor(int64) = ( 01 00 ff ff ff ff ff ff ff ff 00 00 )
+  .method public specialname rtspecialname
+          instance void  .ctor() cil managed
+  {
+    // Code size       7 (0x7)
+    .maxstack  8
+    IL_0000:  ldarg.0
+    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
+    IL_0006:  ret
+  } // end of method C::.ctor
+} // end of class C
+                ]]>
+
+            Dim source =
+<compilation>
+    <file name="attr.vb"><![CDATA[
+Public Class D
+    Shared Sub Main()
+        System.Console.WriteLine(C.F.Ticks)
+    End Sub
+End Class
+]]>
+    </file>
+</compilation>
+
+            ' Using the native compiler, this code outputs 621558279390000000
```

Comment:
Are you sure you didn't try native compiler against a field with two DateTimeConstantAttributes?